pkg/hash/hash_test.go

```diff
@@ -82,7 +101,13 @@ func TestTierHashMatches(t *testing.T) {
 				ClusterResources: &toolchainv1alpha1.NSTemplateTierClusterResources{
 					TemplateRef: "base1ns-clusterresources-e0e1f34-e0e1f34",
 				},
+				SpaceRoles: map[string]toolchainv1alpha1.NSTemplateTierSpaceRole{
+					"admin": {
+						TemplateRef: "base1ns-admin-123456abc",
+					},
+				},
 			},
+			Status: Status,
 		}
```

Comment:
for the purpose of the unit-test, there is no need to define the spec of the NSTemplateTier, rather the opposite - not defining it it will ensure that the spec is not used for the calculation of the hash:
```go
		tier := &toolchainv1alpha1.NSTemplateTier{
			Status: toolchainv1alpha1.NSTemplateTierStatus{
				Revisions: map[string]string{
					"base1ns-dev-aeb78eb-aeb78eb":              "base1ns-dev-aeb78eb-aeb78eb",
					"base1ns-clusterresources-e0e1f34-e0e1f34": "base1ns-clusterresources-e0e1f34-e0e1f34",
					"base1ns-admin-123456abc":                  "base1ns-admin-123456abc",
				},
			},
		}
```

Reply:
you are right, removed the spec part